t/t2717-python-cli-plugins.t

```diff
@@ -0,0 +1,62 @@
+#!/bin/sh
+
+test_description='Test command line plugin interface'
+
+. $(dirname $0)/sharness.sh
+
+export FLUX_CLI_PLUGINPATH=${SHARNESS_TEST_SRCDIR}/cli-plugins/
+
+test_under_flux 4 job
+
+test_expect_success 'flux-alloc: base --help message includes plugin options' '
+  flux run --help | grep -e "Options provided by plugins:" -e "--amd-gpumode"
+'
+test_expect_success 'flux-alloc: a job that does not provide plugins can run' '
+  flux alloc -N1 hostname
+'
+test_expect_success 'flux-alloc: job that calls a plugin has jobspec set properly' '
+  flux alloc --ex-match-policy=firstnodex -N1 --dry-run > jobspec1.json &&
+  test $(jq -r .attributes.system.fluxion_match_policy jobspec1.json) = "firstnodex"
+'
+test_expect_success 'flux-alloc: job with invalid key to plugin is rejected' '
+  test_must_fail flux alloc --ex-match-policy=junkpolicy -N1 echo hello 2> err.out &&
+  grep "flux-alloc: ERROR: Invalid option" err.out
+'
+test_expect_success 'flux-alloc: job with preinit has config set accordingly' '
+  flux alloc --amd-gpumode=TPX -N1 flux config get > config.json && 
+  test $(jq -r .resource.rediscover config.json) = "true"
+'
+test_expect_success 'flux-alloc: multiple valid plugin options are accepted' '
+	flux alloc -N1 --amd-gpumode=TPX --ex-match-policy=firstnodex \
+		flux config get > config2.json &&
+	test $(jq -r .resource.rediscover config2.json) = "true" && 
+	test $(jq -r ".[\"sched-fluxion-resource\"][\"match-policy\"] == \"firstnodex\"" config2.json) = "true"
+'
+test_expect_success 'flux-alloc: plugin without a prefix behaves appropriately' '
+	test_must_fail flux alloc -N1 --multi-user hostname 2> err0.out &&
+  grep "Can only use --multi-user within multi-user instance" err0.out
+'
+test_expect_success 'flux-alloc: job with invalid plugin key is rejected outright' '
+  test_must_fail flux alloc -N1 --ex-junk --ex-notthis=one -N 1 echo hello 2> out2.err &&
+  grep "flux alloc: error: unrecognized arguments" out2.err
+'
+test_expect_success 'flux-run: validate-only plugins are properly validated' '
+  test_must_fail flux run -o signal=foo hostname 2> out3.err &&
+  grep "expected int or mapping got <class" out3.err &&
+  test_must_fail flux run -o signal.foo hostname 2> out4.err &&
+  grep "unsupported keys" out4.err
+'
+test_expect_success 'flux-alloc: new plugin with the same name/dest as another fails' '
+cat <<-EOF > ../cli-plugins/doubleplugin.py && 
+from flux.cli.plugin import CLIPlugin 
+class BadPlugin(CLIPlugin):
+    def __init__(self, prog):
+        super().__init__(prog)
+        self.add_option("--gpumode", help="I want to mess up the gpumode")
+EOF
+  test_must_fail flux run -N1 --dry-run sleep inf 2> out5.err &&
+  grep "conflicts with another option" out5.err &&
+  rm ../cli-plugins/doubleplugin.py
+'
+
+test_done
```

Comment:
It looks like the indentation might be a little inconsistent (at least looking at it on GitHub). Is it possible some indentations are using spaces while others are using tabs?